docs/docker/example.md

```diff
@@ -0,0 +1,132 @@
+# Example of creating a development environment with Docker
+
+This project come with a set of small projects that are bundled together by using Docker.
+
+## Dockerfile
+
+Using an image in a raw way can be useful for quick prototyping. However sometimes when you need to do more things internally e.g installing other packages or running command that need to be done before launching your project. a Dockerfile allow you to add more process overhead on top of a based image.
+
+### Example with the API
+
+In this project the API is made with Go a programming language made by Google. For this project we use the ```alpine``` image. Alpine is a lightweight linux environment. In order to run our API. Our API need some third party plugins.
+
+In order to get these third party plugins we will need to run the command ```go get``` at some point... However ```go get``` need ```git``` to be installed **which is not the cased** on this image.
+
+```
+This can be a good way to make a custom image so we can add git !
+```
+
+So let's create a Dockerfile
+
+#### Dockerfile
+
+A Dockerfile always start from a base image therefore we will use the keyword *FROM*
+
+```dockerfile
+FROM image_name:version
+```
+
+Dockerfile has a set of instruction available such as ADD, RUN, CMD... I suggest you to take a look at the list of instruction available here [Dockerfile instruction](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#dockerfile-instructions)
```

Comment:
Dockerfile has a set of instructions available such as ADD, RUN, CMD... I suggest you take a look at the list [here](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#dockerfile-instructions)

Reply:
Done